### ISSUE TITLE
Dataplane V2

### ISSUE THREAD

**Issue by richiefrich** [terraform-google-modules/terraform-google-kubernetes-engine#1714]
### TL;DR

When I try to enble dataplane v2 nothing changes on the cluster, or even from a new cluster it is not enabled. 

`datapath_provider = ADVANCED_DATAPATH`

I have read you need to not use Network Policy so I also disabled this. 
I am using the lastest provider versions. 

`hashicorp/google = 4.78.0
hashicorp/google-beta = 4.78.0`

Also using version 27.0.0 of the terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant module.

### Expected behavior

That it would enable dataplane v2

### Observed behavior

The dataplane v2 is not enabled from a new cluster or an upgrade.

### Terraform Configuration

```hcl
module "primary-cluster" {
  source                              = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster-update-variant"
  version                             = "27.0.0"
  project_id                          = var.project_id
  name                                = var.primary_cluster
  region                              = var.primary_region
  regional                            = var.is_regional
  zones                               = var.zones
  network                             = var.vpc_name
  subnetwork                          = var.primary_subnet
  ip_range_pods                       = var.primary_pods
  ip_range_services                   = var.primary_services
  cluster_resource_labels             = { "mesh_id" : "proj-${data.google_project.project.number}" }
  datapath_provider                   = var.datapath_provider
  cluster_dns_provider                = var.cluster_dns_provider
  cluster_dns_scope                   = var.cluster_dns_scope
  add_cluster_firewall_rules          = var.add_cluster_firewall_rules
  firewall_inbound_ports              = var.firewall_inbound_ports
  enable_binary_authorization         = var.is_enable_binary_authorization
  grant_registry_access               = var.grant_registry_access
  http_load_balancing                 = var.is_http_load_balancing
  horizontal_pod_autoscaling          = var.is_horizontal_pod_autoscaling
  network_policy                      = var.is_network_policy
  enable_private_endpoint             = var.is_private_endpoint
  enable_private_nodes                = var.is_private_nodes
  master_ipv4_cidr_block              = var.master_ipv4_cidr_block
  istio                               = var.istio
  cloudrun                            = var.cloudrun
  dns_cache                           = var.dns_cache
  release_channel                     = "REGULAR"
  remove_default_node_pool            = var.is_remove_default_node_pool
  master_global_access_enabled        = var.is_master_global_access_enabled
  logging_service                     = var.logging_service
  monitoring_service                  = var.monitoring_service
  enable_network_egress_export        = var.enable_network_egress_export
  enable_resource_consumption_export  = var.enable_resource_consumption_export
  resource_usage_export_dataset_id    = "${var.cust_env}_gke_logs_dataset"
  enable_vertical_pod_autoscaling     = var.enable_vertical_pod_autoscaling
  maintenance_recurrence              = var.maintenance_recurrence
  maintenance_start_time              = var.maintenance_start_time
  maintenance_end_time                = var.maintenance_end_time
  notification_config_topic           = var.gke_pubsub_topic
  enable_intranode_visibility         = var.enable_intranode_visibility
  enable_shielded_nodes               = var.enable_shielded_nodes

  master_authorized_networks          = var.master_authorized_networks
  node_pools                          = var.node_pools
  node_pools_oauth_scopes             = var.node_pools_oauth_scopes
  node_pools_labels                   = var.node_pools_labels
  node_pools_metadata                 = var.node_pools_metadata
  node_pools_tags                     = var.node_pools_tags
}


  add_cluster_firewall_rules                = false
  cloudrun                                  = false
  cluster_dns_provider                      = "CLOUD_DNS"
  cluster_dns_scope                         = "CLUSTER_SCOPE"
  datapath_provider                         = "ADVANCED_DATAPATH"
  dns_cache                                 = false
  enable_cni                                = true
  enable_fleet_registration                 = false
  enable_intranode_visibility               = true
  enable_mesh_feature                       = true
  enable_network_egress_export              = true
  enable_resource_consumption_export        = true
  enable_shielded_nodes                     = true
  enable_vertical_pod_autoscaling           = true
  firewall_inbound_ports                    = ["443", "8443", "9443", "15017"]
  gke_pubsub_topic                          = dependency.pubsub.outputs.gke_pubsub_topic
  grant_registry_access                     = true
  http_load_balancing                       = true
  ip_cidr_range_01                          = "10.128.0.0/20"
  is_auto_repair                            = true
  is_autoscaling                            = true
  is_auto_upgrade                           = true
  is_enable_binary_authorization            = true
  is_horizontal_pod_autoscaling             = true
  is_http_load_balancing                    = true
  is_master_global_access_enabled           = true
  is_network_policy                         = false
  is_private_endpoint                       = false
  is_private_nodes                          = true
  is_regional                               = false
  is_remove_default_node_pool               = true
  istio                                     = false
  jumphost_ip                               = dependency.jumphost.outputs.jumphost_ip
  logging_service                           = "logging.googleapis.com/kubernetes"
  machine_type                              = "n1-standard-4"
  machine_type_core                         = "n1-standard-4"
  machine_type_message_bus                  = "n1-standard-4"
  machine_type_notifications                = "n1-standard-4"
  machine_type_cron                         = "n1-standard-4"
  maintenance_end_time                      = "2022-01-01T23:30:00Z"
  maintenance_recurrence                    = "FREQ=DAILY"
  maintenance_start_time                    = "2022-01-01T11:00:00Z"
```

### Terraform Version

```sh
Terraform v1.4.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/google v4.78.0
+ provider registry.terraform.io/hashicorp/google-beta v4.78.0
+ provider registry.terraform.io/hashicorp/helm v2.10.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.19.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/mongodb/mongodbatlas v1.8.2
```

### Additional information

If I am just missing some option pease let me know!

**Comment by volver-13** [comment]
_GKE Dataplane V2 can only be enabled when creating a new cluster. Existing clusters cannot be upgraded to use GKE Dataplane V2._

Source: https://cloud.google.com/kubernetes-engine/docs/concepts/dataplane-v2

Can you please describe your cluster with:
```
gcloud container clusters describe <cluster_name> --location <location> | grep datapathProvider
```

**Comment by richiefrich** [comment]
@volver-13  
I get nothing back
`$ gcloud --project=ss-pp-gcp-asm  container clusters describe uat-01 --location us-central1-a   | grep datapathProvider
 gcloud --project=ss-pp-gcp-asm  container clusters list
NAME    LOCATION       MASTER_VERSION  MASTER_IP     MACHINE_TYPE   NODE_VERSION    NUM_NODES  STATUS
uat-01  us-central1-a  1.27.3-gke.100  34.71.155.63  n1-standard-4  1.27.3-gke.100  2          RUNNING`

**Comment by richiefrich** [comment]
This is now working.
